### PR TITLE
Add features

### PR DESCRIPTION
**Pull request by rhysgt**
- Make EDX data accessible from EBSD map object
- Change `component` to a property of map
- Allow changing IPF map background colour
- Update CHANGELOG.md accordingly

### CODE REVIEW

File: defdap/file_readers.py [hunk 1]
```diff
@@ -306,15 +309,19 @@ def parseLine(line: str, groupDict: Dict) -> None:
...
             ))
-
+ 
         # Deal with EDX data
+        edx_fields = {}
         if 'EDX Windows' in metadata:
             edx_dict = metadata['EDX Windows']
             num_edx = int(edx_dict['Count'])
+            self.loadedMetadata['EDX Window Count'] = num_edx
             edx_fields = {}
```
Comment (mikesmic) [inline defdap/file_readers.py:319]: Not needed, defined above

File: defdap/hrdic.py [hunk 2]
```diff
@@ -175,17 +175,21 @@ def __init__(self, path, fname, dataType=None):
...
+    @property
+    def component(self):
+        # Dictionary references to all map components
+        components = {'f11': self.f11, 'f12': self.f12, 'f21': self.f21, 'f22': self.f22,
+                         'e11': self.e11, 'e12': self.e12, 'e22': self.e22,
+                         'eMaxShear': self.eMaxShear,
+                         'x_map': self.x_map, 'y_map': self.y_map}
+
+        return components
+
```
Comment (mikesmic) [inline defdap/hrdic.py:192]: I don't like the idea of doing this, it means you could never add new components to the map for plotting (eventual goal for this). If there are issues with the references changing after your masking process you should try and apply the mask to the data in-place or change update the specific entries in the dict that the mask is applied to

**Review by mikesmic — CHANGES_REQUESTED** [review]
I don't want to change the component dictionary to a property as then it cannot be updated at runtime. We need to find a different way to address the issue this is fixing

File: defdap/hrdic.py (hunk 2 above)
Comment (rhysgt) [reply defdap/hrdic.py]: Fixed in 8e25626

Comment (rhysgt) [reply defdap/hrdic.py]: We could consider removing the e11, e22… variables and defining these within the component dictionary in the first place actually which would tidy things up further.

Comment (mikesmic) [reply defdap/hrdic.py]: Yeh, fine with that. We should move away from storing data directly as attributes on the objects and instead store in a dictionary with metadata included as well, one day I might make all the changes and make it more streamlined to work with data created at runtime

File: defdap/file_readers.py (hunk 1 above)
Comment (rhysgt) [reply defdap/file_readers.py:317]: Have refactored to remove duplication in b9ce103. Let me know if any more changes are required. Have tested on OxfordBinary EBSD data with/without EDX and OxfordText data.

File: defdap/hrdic.py (hunk 2 above)
Comment (rhysgt) [reply defdap/hrdic.py]: I would propose a format similar to this, don't know what you think?

```
        # Dictionary to store map components
        self.component={}

        # Deformation gradient
        self.component['f11']= xDispGrad[1] + 1
        self.component['f22'] = yDispGrad[0] + 1
        self.component['f12'] = xDispGrad[0]
        self.component['f21'] = yDispGrad[1]

        # Green strain
        self.component['e11'] = xDispGrad[1] + \
                   0.5*(xDispGrad[1]*xDispGrad[1] + yDispGrad[1]*yDispGrad[1])
        self.component['e22'] = yDispGrad[0] + \
                   0.5*(xDispGrad[0]*xDispGrad[0] + yDispGrad[0]*yDispGrad[0])
        self.component['e12'] = 0.5*(xDispGrad[0] + yDispGrad[1] +
                        xDispGrad[1]*xDispGrad[0] + yDispGrad[1]*yDispGrad[0])

        # max shear component
        self.component['eMaxShear'] = np.sqrt(((self.e11 - self.e22) / 2.)**2 + self.e12**2)

```

**Comment by mikesmic** [comment]
Sorry for the delay. Looks good to me! Happy for you to merge if you have checked it can still load CPR files with and without EDX data included. We can think about what to do with map components at a later time, it would be nice to have something implemented in the base map class with a general interface for adding, removing, accessing and modifying map data components.

**Review by mikesmic — APPROVED** [review]

**Pull request merged**